File: src/types/autoMode.ts
/**
 * Auto-Mode 类型定义
 *
 * 用于自动模式配置的 TypeScript 类型
 */

/**
 * 自动模式配置
 */
export interface AutoModeConfig {
  /** 允许规则列表 */
  allow: string[];
  /** 拒绝规则列表（软拒绝，需确认） */
  soft_deny: string[];
  /** 环境配置 */
  environment: string[];
}

/**
 * 自动模式默认配置
 */
export interface AutoModeDefaults {
  /** 默认允许规则 */
  allow: string[];
  /** 默认拒绝规则 */
  soft_deny: string[];
  /** 默认环境配置 */
  environment: string[];
}

/**
 * 规则类型
 */
export type RuleType = 'allow' | 'softDeny';

/**
 * 规则分类
 */
export interface RuleCategory {
  id: string;
  name: string;
  description: string;
  rules: string[];
}

File: src-tauri/src/models/auto_mode.rs
//! Auto-Mode 数据模型
//!
//! 用于 Claude CLI 自动模式配置的数据结构

use serde::{Deserialize, Serialize};

/// 自动模式配置
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AutoModeConfig {
    /// 允许规则列表
    pub allow: Vec<String>,
    /// 拒绝规则列表（软拒绝，需确认）
    pub soft_deny: Vec<String>,
    /// 环境配置
    pub environment: Vec<String>,
}

/// 自动模式默认配置
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AutoModeDefaults {
    /// 默认允许规则
    pub allow: Vec<String>,
    /// 默认拒绝规则
    pub soft_deny: Vec<String>,
    /// 默认环境配置
    pub environment: Vec<String>,
}
